apis/aws/v1alpha1/role_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// RoleParameters are the configurable fields of a Role.
+type RoleParameters struct {
+	Backend               string   `json:"authBackend" binding:"required"`
+	CredentialType        string   `json:"credentialType" binding:"required"`
+	IamRolesArn           []string `json:"iamRoles" binding:"required"`
+	PoliciesArn           []string `json:"policiesArn" binding:"required"`
+	PolicyDocument        string   `json:"policyDocument" binding:"required"`
+	IamGroups             []string `json:"iamGroups" binding:"required"`
+	UserPath              string   `json:"userPath" binding:"required"`
+	PermissionBoundaryArn string   `json:"permissionsBoundaryArn" binding:"required"`
+	DefaultStsTTL         int      `json:"defaultStsTtl" binding:"required"`
+	MaxStsTTL             int      `json:"maxStsTtl" binding:"required"`
```

Comment:
Hey Brenno, I would add docs to all the fields here. This documentation will be put into the CRDs generated. Also, I think the `binding:"required"` is useless in this scenario. Just make sure you add the optional fields as pointers and put the `// +optional` comment on top of it. For those things, you may check how aws-provider does for example or even check my branch.